### ISSUE TITLE
BlockCancun commit had to be reverted after switch to IPA

### ISSUE THREAD

**Issue by gballet** [gballet/go-verkle#121]
After the switch to IPA, the changes introduced in 688952b18b56dcdcf1bb90ab7646b1a8f330214f seem to no longer work.

In particular, the following command

```
rm -rf .verkle && go build -tags bignum_kilic ./cmd/geth/... && ./geth init genesis_verkle.json && ./geth --datadir=.verkle --nodiscover --mine --miner.threads=1 --http --http.corsdomain="*" --networkid 86 --http.rpcprefix="/verkle" --miner.etherbase=0x617661d148A52bef51a268C728b3A21B58f94306
```

fails during the `init` stage, because it tries to (de)serialize a verkle node as a regular MPT node. Not sure why this failed, but I had to revert it in d95451bdba7a4ebc825e7423e01ad0f0af4ffe6d in order to be able to run it.

**Comment by jwasinger** [comment]
That's weird.  It works fine for me.  Can you share the genesis block that failed?  This is what I used:
```
{
  "config": {
    "chainId": 86,
    "homesteadBlock": 0,
    "daoForkBlock": 0,
    "daoForkSupport": true,
    "eip150Block": 0,
    "eip150Hash": "0x2086799aeebeae135c246c65021c82b4e15a2c451340993aacfd2751886514f0",
    "eip155Block": 0,
    "eip158Block": 0,
    "byzantiumBlock": 0,
    "constantinopleBlock": 0,
    "petersburgBlock": 0,
    "istanbulBlock": 0,
    "muirGlacierBlock": 0,
    "berlinBlock": 0,
    "londonBlock": 0,
    "cancunBlock": 0,
    "ethash": {}
  },
  "nonce": "0x56",
  "timestamp": "0x0",
  "extraData": "0x",
  "gasLimit": "0x2fefd8",
  "difficulty": "0x1",
  "mixHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "coinbase": "0x0000000000000000000000000000000000000000",
  "alloc": {
    "34a600a929c439fcc9fd87bf493fea453add3d5f": {
      "balance": "0x40000000000000000000"
    },
    "28a03b1dd1e2155cb230f27c8251c47338c36e66": {
      "balance": "0x40000000000000000000"
    }
  },
  "number": "0x0",
  "gasUsed": "0x0",
  "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "baseFeePerGas": null
}
```

**Comment by jwasinger** [comment]
It fails on client startup for me.  not `init`.

**Comment by gballet** [comment]
Indeed my genesis JSON is much smaller:

```
{
   "alloc": {},
   "coinbase" : "0x0000000000000000000000000000000000000000",
   "config" : {
      "byzantiumBlock" : 0,
      "chainId" : 56,
      "constantinopleBlock" : 0,
      "eip150Block" : 0,
      "eip155Block" : 0,
      "eip158Block" : 0,
      "homesteadBlock" : 0,
      "istanbulBlock" : 0,
      "petersburgBlock" : 0,
      "cancunBlock": 0
   },
   "difficulty" : "0x01",
   "extraData" : "",
   "gasLimit" : "0x86",
   "mixhash" : "0x0000000000000000000000000000000000000000000000000000000000000000",
   "nonce" : "0x0000000000220720",
   "parentHash" : "0x0000000000000000000000000000000000000000000000000000000000000000",
   "timestamp" : "0x00"
}
```

Not sure why it fails the way it does, though.

**Comment by gballet** [comment]
Fixed in afd21668aa868eab022c38223dde33a85560f421. Closing.
